src/PrestaShopBundle/ApiPlatform/Provider/QueryProvider.php

```diff
@@ -28,33 +28,105 @@
         private readonly CommandBusInterface $queryBus,
+        private readonly iterable $converters,
         private readonly Serializer $apiPlatformSerializer
     ) {
     }
 
+    /**
+     * @throws \ReflectionException
+     * @throws \Exception
+     * @throws ExceptionInterface
+     */
     public function provide(Operation $operation, array $uriVariables = [], array $context = [])
     {
-        $extraProperties = $operation->getExtraProperties();
+        $queryClass = $operation->getExtraProperties()['query'] ?? null;
+        $filters = $context['filters'] ?? [];
+        $uriVariables = array_merge($uriVariables, $filters);
 
-        $query = $extraProperties['query'] ?? null;
-
-        if (null === $query) {
+        if (null === $queryClass) {
             throw new NoExtraPropertiesFoundException();
         }
 
-        $queryResult = $this->queryBus->handle(new $query(...$uriVariables));
+        $query = $this->apiPlatformSerializer->denormalize($uriVariables, $queryClass);
+
+        //Try to call setter on additional query params
+        if (count($uriVariables)) {
+            foreach ($uriVariables as $param => $value) {
+                if ($reflectionMethod = $this->findSetterMethod($param, $queryClass)) {
+                    $methodParameter = $reflectionMethod->getParameters()[0];
+                    if ($methodParameter->getType() instanceof \ReflectionNamedType && $methodParameter->getType()->getName() !== gettype($value)) {
+                        $value = $this->findConverter($methodParameter->getType()->getName())->convert($value);
+                    }
+                    $reflectionMethod->invoke($query, $value);
+                }
+            }
+        }
+
+        $queryResult = $this->queryBus->handle($query);
         $normalizedQueryResult = $this->apiPlatformSerializer->normalize($queryResult);
 
+        if ($operation instanceof CollectionOperationInterface) {
+            foreach ($normalizedQueryResult as $key => $result) {
+                $normalizedQueryResult[$key] = $this->apiPlatformSerializer->denormalize($result, $operation->getClass());
+            }
+
+            return $normalizedQueryResult;
+        }
+
         return $this->apiPlatformSerializer->denormalize($normalizedQueryResult, $operation->getClass());
     }
+
+    /**
+     * @param $type
+     *
+     * @return ConverterInterface
+     *
+     * @throws \Exception
+     */
+    private function findConverter($type): ConverterInterface
+    {
+        foreach ($this->converters as $converter) {
+            if ($converter->supports($type)) {
+                return $converter;
+            }
+        }
+
+        throw new \Exception(sprintf('Converter for type %s not found', $type));
+    }
+
+    /**
+     * @param $propertyName
+     * @param $queryClass
+     *
+     * @return false|\ReflectionMethod
+     */
+    private function findSetterMethod($propertyName, $queryClass): bool|\ReflectionMethod
```

Comment:
This logic looks a lot like what PropertyAccessor already does so I wonder if it's still needed since it's used in the serializer and especially on the `ObjectNormalizer`

Reply:
Maybe we can simplify it if it's possible in this issue  https://github.com/PrestaShop/PrestaShop/issues/32805